lib/ProductOpener/GS1.pm

```diff
@@ -852,6 +852,60 @@ sub assign_field($results_ref, $target_field, $target_value) {
 }
 
 
+sub nutrient_per_quantity ($type, $per, $results_ref, $nid=undef, $nutrient_detail_ref=undef) {
```

Comment:
We should never call the function without arguments for $nid and $nutrient_detail_ref, so we should not set default values for them.